packages/client/tests/functional/issues/8832/tests.ts

```diff
@@ -0,0 +1,202 @@
+        where: {
+          id: {
+            in: ids,
+          },
+        },
+      })
+      return tags
+    }
+
+    async function getTagsFindManyInclude(): Promise<unknown[]> {
+      const tags = await prisma.tag.findMany({
+        include: {
+          posts: true,
+        },
+      })
+      return tags
+    }
+
+    async function clean() {
+      const cleanPrismaPromises = [prisma.tagsOnPosts.deleteMany(), prisma.post.deleteMany(), prisma.tag.deleteMany()]
+      await prisma.$transaction(cleanPrismaPromises)
+    }
+
+    describe('issue #8832', () => {
+      beforeEach(async () => {
+        await clean()
+      }, 10_000)
+
+      test('should succeed when "in" has 32766 ids', async () => {
+        const n = 32766
+        const ids = await createTags(n)
+        const tags = await getTagsFindManyIn(ids)
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "include" involves 32766 records', async () => {
+        const n = 32766
+        await createTags(n)
+        const tags = await getTagsFindManyInclude()
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "in" has 32767 ids', async () => {
+        const n = 32767
+        const ids = await createTags(n)
+        const tags = await getTagsFindManyIn(ids)
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "include" involves 32767 records', async () => {
+        const n = 32767
+        await createTags(n)
+        const tags = await getTagsFindManyInclude()
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "in" has 32768 ids', async () => {
+        const n = 32768
+        const ids = await createTags(n)
+        const tags = await getTagsFindManyIn(ids)
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "include" involves 32768 records', async () => {
+        const n = 32768
+        await createTags(n)
+        const tags = await getTagsFindManyInclude()
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "in" has 65536 ids', async () => {
+        const n = 65536
+        const ids = await createTags(n)
+        const tags = await getTagsFindManyIn(ids)
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "include" involves 65536 records', async () => {
+        const n = 65536
+        await createTags(n)
+        const tags = await getTagsFindManyInclude()
+
+        expect(tags.length).toBe(n)
+      })
+
+      /**
+       * Highlight scenarios of the query chunking logic behavior depending on how the bind variables are counted
+       * in the query filters.
+       */
+      describe('unhandled filters', () => {
+        test('should fail with `value too large to transmit` when "in" is repeated at least twice and "n" is $QUERY_BATCH_SIZE', async () => {
+          expect.assertions(3)
+          const n = 32766
```

Comment:
Since `QUERY_BATCH_SIZE` is not defined (can't find it), how do we know that `n === QUERY_BATCH_SIZE` here?

Reply:
I _think_ what Alberto means here is that n is the same as the default chunk size for PostgreSQL (which can be modifed via `QUERY_BATCH_SIZE`).

Reply:
`$QUERY_BATCH_SIZE` indeed indicates the environment variable `QUERY_BATCH_SIZE`, which influences the default values in [here](https://github.com/prisma/prisma-engines/blob/main/query-engine/connectors/sql-query-connector/src/sql_info.rs).

Reply:
`QUERY_BATCH_SIZE` is not used in this test.

Reply:
Updated test name as:

> should fail with `Assertion violation` when "in" is repeated at least twice and "n" is the default value of $QUERY_BATCH_SIZE